File: sandbox-runner/patterns/early-return/java/src/main/java/Starter.java
import io.temporal.api.enums.v1.WorkflowIdConflictPolicy;
import io.temporal.client.UpdateOptions;
import io.temporal.client.WithStartWorkflowOperation;
import io.temporal.client.WorkflowClient;
import io.temporal.client.WorkflowOptions;
import io.temporal.client.WorkflowStub;
import io.temporal.client.WorkflowUpdateHandle;
import io.temporal.client.WorkflowUpdateStage;
import io.temporal.serviceclient.WorkflowServiceStubs;

public class Starter {
    public static void main(String[] args) throws Exception {
        WorkflowServiceStubs service = WorkflowServiceStubs.newLocalServiceStubs();
        WorkflowClient client = WorkflowClient.newInstance(service);

        String transactionId = Shared.WORKFLOW_ID_PREFIX + "-" + System.currentTimeMillis();
        Shared.TransactionRequest req = new Shared.TransactionRequest(100.0, "USD");

        TransactionWorkflow workflow = client.newWorkflowStub(
                TransactionWorkflow.class,
                WorkflowOptions.newBuilder()
                        .setTaskQueue(Shared.TASK_QUEUE)
                        .setWorkflowId(transactionId)
                        .setWorkflowIdConflictPolicy(
                                WorkflowIdConflictPolicy.WORKFLOW_ID_CONFLICT_POLICY_FAIL)
                        .build());

        long t0 = System.currentTimeMillis();
        WorkflowUpdateHandle<Shared.Transaction> updateHandle =
                WorkflowClient.startUpdateWithStart(
                        workflow::returnInitResult,
                        UpdateOptions.<Shared.Transaction>newBuilder()
                                .setWaitForStage(WorkflowUpdateStage.COMPLETED)
                                .build(),
                        new WithStartWorkflowOperation<>(workflow::processTransaction, req));

        Shared.Transaction tx = updateHandle.getResultAsync().get();
        System.out.printf("Early return after %dms: %s (status=%s)%n",
                System.currentTimeMillis() - t0, tx.id(), tx.status());

        WorkflowStub stub = WorkflowStub.fromTyped(workflow);
        Shared.Transaction finalTx = stub.getResult(Shared.Transaction.class);
        System.out.printf("Workflow completed after %dms: %s (status=%s)%n",
                System.currentTimeMillis() - t0,
                finalTx == null ? "null" : finalTx.id(),
                finalTx == null ? "n/a" : finalTx.status());
        System.out.println("Open the Temporal UI and search for '" + transactionId
                + "' to see the history.");

        System.exit(0);
    }
}

File: sandbox-runner/patterns/early-return/go/starter.go
package main

import (
	"context"
	"fmt"
	"log"
	"time"

	enumspb "go.temporal.io/api/enums/v1"
	"go.temporal.io/sdk/client"
)

func main() {
	c, err := client.Dial(client.Options{HostPort: "localhost:7233"})
	if err != nil {
		log.Fatalln("Unable to create client:", err)
	}
	defer c.Close()

	ctx := context.Background()
	transactionID := fmt.Sprintf("%s-%d", WorkflowIDPrefix, time.Now().UnixMilli())
	req := TransactionRequest{Amount: 100, Currency: "USD"}

	startOp := c.NewWithStartWorkflowOperation(
		client.StartWorkflowOptions{
			ID:                       transactionID,
			TaskQueue:                TaskQueue,
			WorkflowIDConflictPolicy: enumspb.WORKFLOW_ID_CONFLICT_POLICY_FAIL,
		},
		TransactionWorkflow,
		req,
	)

	t0 := time.Now()
	updateHandle, err := c.UpdateWithStartWorkflow(ctx, client.UpdateWithStartWorkflowOptions{
		StartWorkflowOperation: startOp,
		UpdateOptions: client.UpdateWorkflowOptions{
			UpdateName:   UpdateName,
			WaitForStage: client.WorkflowUpdateStageCompleted,
		},
	})
	if err != nil {
		log.Fatalln("UpdateWithStartWorkflow failed:", err)
	}

	var tx Transaction
	if err := updateHandle.Get(ctx, &tx); err != nil {
		log.Fatalln("Update result failed:", err)
	}
	fmt.Printf("Early return after %dms: %s (status=%s)\n",
		time.Since(t0).Milliseconds(), tx.ID, tx.Status)

	we, err := startOp.Get(ctx)
	if err != nil {
		log.Fatalln("Workflow start failed:", err)
	}
	var final Transaction
	if err := we.Get(ctx, &final); err != nil {
		log.Fatalln("Workflow result failed:", err)
	}
	fmt.Printf("Workflow completed after %dms: %s (status=%s)\n",
		time.Since(t0).Milliseconds(), final.ID, final.Status)
	fmt.Printf("Open the Temporal UI and search for '%s' to see the history.\n", transactionID)
}
